cns/restserver/ipam.go

```diff
@@ -526,42 +549,73 @@ func (service *HTTPRestService) AssignDesiredIPConfig(podInfo cns.PodInfo, desir
 			default:
 				return podIpInfo, errors.Errorf("[AllocateDesiredIPConfig] Desired IP is not available %+v", ipConfig)
 			}
-			err := service.populateIPConfigInfoUntransacted(ipConfig, &podIpInfo)
+			err := service.populateIPConfigInfoUntransacted(ipConfig, &podIpInfo[0])
 			return podIpInfo, err
 		}
 	}
 	return podIpInfo, fmt.Errorf("Requested IP not found in pool")
 }
 
-func (service *HTTPRestService) AssignAnyAvailableIPConfig(podInfo cns.PodInfo) (cns.PodIpInfo, error) {
+// ryand setting to default dualstack mode for testing while waiting to get pod labels - will change
```

Comment:
this logic needs to change so that it gives out one IP from each available NC. There is no need to be concerned about v4 / v6